### PR TITLE
Rename SDLmain to SDL_main and SDLtest to SDL_test for consistency with other SDL libraries

### PR DESCRIPTION
**Pull request by slouken**

### CODE REVIEW

File: docs/README-cmake.md [hunk 1]
```diff
@@ -87,8 +87,8 @@ The following components are available, to be used as an argument of `find_packa
 |----------------|--------------------------------------------------------------------------------------------|
 | SDL3           | The SDL3 shared library, available through the `SDL3::SDL3` target [^SDL_TARGET_EXCEPTION] |
 | SDL3-static    | The SDL3 static library, available through the `SDL3::SDL3-static` target                  |
-| SDL3main       | The SDL3main static library, available through the `SDL3::SDL3main` target                 |
-| SDL3test       | The SDL3test static library, available through the `SDL3::SDL3test` target                 |
+| SDL3_main       | The SDL3_main static library, available through the `SDL3::SDL3_main` target                 |
+| SDL3_test       | The SDL3_test static library, available through the `SDL3::SDL3_test` target                 |
```
Comment (madebr) [inline docs/README-cmake.md:91]: I removed one space from the first column, and two from the second.
```suggestion
| SDL3_main      | The SDL3_main static library, available through the `SDL3::SDL3_main` target               |
| SDL3_test      | The SDL3_test static library, available through the `SDL3::SDL3_test` target               |

```

File: docs/README-migration.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ CFLAGS += $(shell pkg-config sdl3 --cflags)
 LDFLAGS += $(shell pkg-config sdl3 --libs)
 ```
 
+The SDLmain and SDLtest libraries have been renamed SDL_main and SDL_test, respectively.
```
Comment (madebr) [inline docs/README-migration.md:31]: I suppose we don't care about the version number here, and in comments?

Comment (slouken) [reply docs/README-migration.md:31]: No, leaving the version off is fine.

Comment (slouken) [reply docs/README-migration.md:31]: Actually, maybe we do want it, since it'll be the name of the library they link with.

**Pull request merged**